flow/layers/layer.h

```diff
@@ -223,15 +223,19 @@ class Layer {
 
     ~AutoCachePaint() { context_.inherited_opacity = sk_paint_.getAlphaf(); }
 
-    void setImageFilter(sk_sp<SkImageFilter> filter) {
-      sk_paint_.setImageFilter(filter);
-      dl_paint_.setImageFilter(DlImageFilter::From(filter));
+    void setImageFilter(const DlImageFilter* filter) {
+      if (!filter) {
+        return;
+      }
+      sk_paint_.setImageFilter(filter->skia_object());
+      dl_paint_.setImageFilter(filter);
       update_needs_paint();
     }
 
     void setColorFilter(const DlColorFilter* filter) {
-      if (!filter)
+      if (!filter) {
```

Comment:
Same idea here - should we clear the CF attribute just in case we eventually use this object in a "set it and later clear it" manner.

Reply:
fixed